enhancements/cloud-integration/azure/azure-workload-identity.md

```diff
@@ -0,0 +1,358 @@
+
+Flags:
+  -h, --help   help for azure
+
+Use "ccoctl azure [command] --help" for more information about a command.
+```
+
+#### Azure CredentialsRequests ServiceAccounts
+
+`CredentialsRequests` for the Azure platform must now list
+[ServiceAccountNames](https://github.com/openshift/cloud-credential-operator/blob/1f7a2602bf8a9ddec5d8fc29f77215697d9e7c07/pkg/apis/cloudcredential/v1/types_credentialsrequest.go#L57-L62)
+in order to for `ccoctl` to be able to create federated credentials
+for an Azure Managed Identity that are associated with the `name` and
+`namespace` of the `ServiceAccount`. Example:
+[aws-ebs-csi-driver-operator](https://github.com/openshift/cluster-storage-operator/blob/f1ddb697afb3c33d6d45936e58fad101abe26f13/manifests/03_credentials_request_aws.yaml#L11-L13).
+
+#### Credentials secret
+
+OpenShift operators currently obtain their long-lived credentials from a config secret with the following format:
+
+```yaml
+apiVersion: v1
+data:
+  azure_client_id: <client id>
+  azure_client_secret: <client secret>
+  azure_region: <region>
+  azure_resource_prefix: <resource group prefix eg. "abutcher-az-t68n4">
+  azure_resourcegroup: <resource group eg. "abutcher-az-t68n4-rg">
+  azure_subscription_id: <subscription id>
+  azure_tenant_id: <tenant id>
+kind: Secret
+type: Opaque
+```
+
+We propose that when utilizing Azure Workload Identity, the credentials secret will contain an `azure_client_id` that is the `clientID` of the Managed Identity provisioned by `ccoctl` for the operator. The `azure_client_secret` key will be absent and instead we can provide the path to the mounted `ServiceAccount` token as an `azure_federated_token_file` key; the path to the mounted token is well
+known and is specified in the operator deployment.
+
+The resource group in which the installer will create infrastructure will not be known when these secrets are generated by `ccoctl` ahead of installation and operators which rely on `azure_resourcegroup` and `azure_resource_prefix` such as the
+[image-registry](https://github.com/openshift/cluster-image-registry-operator/blob/8556fd48027f89e19daad36e280b60eb93d012d4/pkg/storage/azure/azure.go#L95-L100) should obtain the resource group details from the cluster `Infrastructure` object instead.
+
+```yaml
+apiVersion: v1
+data:
+  azure_client_id: <client id>
+  azure_federated_token_file: <path to mounted service account token, eg. "/var/run/secrets/openshift/serviceaccount/token">
+  azure_region: <region>
+  azure_subscription_id: <subscription id>
+  azure_tenant_id: <tenant id>
+kind: Secret
+type: Opaque
+```
+
+#### Creating workload identity clients in operators
+
+In order to create Azure clients which utilize a `ClientAssertionCredential`, operators must update to version `>= v1.3.0-beta.4` of the azidentity package within [azure-sdk-for-go](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity@v1.3.0-beta.4). Ahead of this work, due to the [end of life
```

Comment:
:warning: Please note that the operator changes required to support workload identity are based on azure-sdk-for-go version `v1.3.0-beta.4`.

Reply:
We should wait to merge operator changes until `v.1.3.0` is out of beta.

Reply:
> We should wait to merge operator changes until v.1.3.0 is out of beta.

why wait if they're ready.  Coming back to update when it's GA, sure.  But I'd hate to see us lose time for fear of trying a beta.